packages/tempus/src/Tempus_TimeStepControl_impl.hpp

```diff
@@ -667,58 +695,65 @@ Teuchos::RCP<TimeStepControl<Scalar> > createTimeStepControl(
-  } else if (tsc->getStepType() == "Variable") {
-     // add TSCS from "Time Step Control Strategy List"
-
-     RCP<ParameterList> tscsPL =
-       Teuchos::sublist(pList, "Time Step Control Strategy", true);
-     // Construct from TSCS sublist
-     std::vector<std::string> tscsLists;
-
-     // string tokenizer
-     tscsLists.clear();
-     std::string str = tscsPL->get<std::string>("Time Step Control Strategy List");
-     std::string delimiters(",");
-     // Skip delimiters at the beginning
-     std::string::size_type lastPos = str.find_first_not_of(delimiters, 0);
-     // Find the first delimiter
-     std::string::size_type pos     = str.find_first_of(delimiters, lastPos);
-     while ((pos != std::string::npos) || (lastPos != std::string::npos)) {
-        // Found a token, add it to the vector
-        std::string token = str.substr(lastPos,pos-lastPos);
-        tscsLists.push_back(token);
-        if(pos==std::string::npos) break;
-
-        lastPos = str.find_first_not_of(delimiters, pos); // Skip delimiters
-        pos = str.find_first_of(delimiters, lastPos);     // Find next delimiter
-     }
-
-     // For each sublist name tokenized, add the TSCS
-     for( auto el: tscsLists){
+  if (tsc->getStepType() == "Variable") {
+    // add TSCS from "Time Step Control Strategy List"
 
-        RCP<ParameterList> pl =
-           Teuchos::rcp(new ParameterList(tscsPL->sublist(el)));
+    RCP<ParameterList> tscsPL =
+      Teuchos::sublist(pList, "Time Step Control Strategy", true);
+    // Construct from TSCS sublist
+    std::vector<std::string> tscsLists;
+
+    // string tokenizer
+    tscsLists.clear();
+    std::string str = tscsPL->get<std::string>("Time Step Control Strategy List");
+    std::string delimiters(",");
+    // Skip delimiters at the beginning
+    std::string::size_type lastPos = str.find_first_not_of(delimiters, 0);
+    // Find the first delimiter
+    std::string::size_type pos     = str.find_first_of(delimiters, lastPos);
+    while ((pos != std::string::npos) || (lastPos != std::string::npos)) {
+      // Found a token, add it to the vector
+      std::string token = str.substr(lastPos,pos-lastPos);
+      tscsLists.push_back(token);
+      if(pos==std::string::npos) break;
 
-        RCP<TimeStepControlStrategy<Scalar>> ts;
+      lastPos = str.find_first_not_of(delimiters, pos); // Skip delimiters
+      pos = str.find_first_of(delimiters, lastPos);     // Find next delimiter
+    }
 
-        // construct appropriate TSCS
-        if(pl->get<std::string>("Name") == "Integral Controller")
-           ts = Teuchos::rcp(new TimeStepControlStrategyIntegralController<Scalar>(pl));
-        else if(pl->get<std::string>("Name") == "Basic VS")
-           ts = Teuchos::rcp(new TimeStepControlStrategyBasicVS<Scalar>(pl));
+    Teuchos::RCP<TimeStepControlStrategy<Scalar> > tscs =
+      Teuchos::rcp(new TimeStepControlStrategyConstant<Scalar>());
```

Comment:
So this here is in the Variable stepping block. I'm not sure I understand the reasoning.

Reply:
Yeah, this coding has changed with my current changes which have not been committed yet.   This is some of the coding that was simplified by going with a TimeStepControl only having a single Strategy.  So I would suggest waiting on my up coming commit.  

How urgent do you need these changes?  I have been talking Eric Cyr about when I can merge this PR, and we are waiting for Jason Gates to return from vacation.  So hopefully next week.

Reply:
Not urging at all